cpp/src/arrow/compute/kernels/scalar_cast_temporal.cc

```diff
@@ -519,6 +521,12 @@ std::shared_ptr<CastFunction> GetDurationCast() {
   // Between durations
   AddCrossUnitCast<DurationType>(func.get());
 
+  // time32/64 -> duration
```

Comment:
I'm not sure I understand what it means to cast from a point in time to a duration. Does it make sense?